test/remote/gateways/remote_stripe_payment_intents_test.rb

```diff
@@ -64,6 +100,160 @@ def test_successful_purchase
   end
 
+  def test_successful_tokenization_google_pay
+    options = {
+      currency: 'GBP'
+    }
+    assert store = @gateway.store(@google_pay, options)
+    assert_equal store.params['token']['card']['tokenization_method'], 'android_pay'
+    assert store.params['token']['id'].start_with?('tok_')
+  end
+
+  def test_successful_store_google_pay
+    options = {
+      currency: 'GBP'
+    }
+    token = @gateway.store(@google_pay, options)
+    assert token.success?
+    assert_equal token.params['token']['card']['tokenization_method'], 'android_pay'
+    stripe_token = token.params['token']['id']
+    auth = @gateway.store(stripe_token, options)
+    assert auth.success?
+  end
+
+  def test_successful_store_apple_pay
+    options = {
+      currency: 'GBP'
+    }
+    token = @gateway.store(@apple_pay, options)
+    assert token.success?
+    assert_equal token.params['token']['card']['tokenization_method'], 'apple_pay'
+    stripe_token = token.params['token']['id']
+    auth = @gateway.store(stripe_token, options)
+    assert auth.success?
+  end
+
+  def test_successful_tokenization_apple_pay
+    options = {
+      currency: 'GBP'
+    }
+    assert store = @gateway.store(@apple_pay, options)
+    assert_equal store.params['token']['card']['tokenization_method'], 'apple_pay'
+    assert store.params['token']['id'].start_with?('tok_')
+  end
+
+  def test_unsuccessful_token_google_pay_with_incorrect_card_number
+    options = {
+      currency: 'GBP'
+    }
+    store = @gateway.store(@incorrect_google_pay, options)
+    assert_equal 'Your card number is incorrect.', store.message
+  end
+
+  def test_unsuccessful_token_google_pay_with_invalid_tokenization_method
+    options = {
+      currency: 'GBP',
+      tokenization_method: 'credit_card'
+    }
+    store = @gateway.store(@google_pay, options)
+    assert_equal "The card number provided is not an tokenized number, but you specified 'cryptogram' which is only valid for tokenized card numbers.", store.message
+    assert_false store.success?
+  end
+
+  def test_unsuccessful_token_google_pay_without_cryptogram
+    options = {
+      currency: 'GBP'
+    }
+    @google_pay.payment_cryptogram = ''
+    store = @gateway.store(@google_pay, options)
+    assert_equal "Cards using 'tokenization_method=android_pay' require the 'cryptogram' field to be set.", store.message
+    assert_false store.success?
+  end
+
+  def test_unsuccessful_token_google_pay_without_month
+    options = {
+      currency: 'GBP'
+    }
+    @google_pay.month = ''
+    store = @gateway.store(@google_pay, options)
+    assert_equal 'Missing required param: card[exp_month].', store.message
+    assert_false store.success?
+  end
+
+  def test_successful_authorize_with_google_pay
+    options = {
+      currency: 'GBP'
+    }
+    store = @gateway.store(@google_pay, options)
+    assert store.success?
+    stripe_token = store.params['token']['id']
+    auth = @gateway.authorize(@amount, stripe_token, options)
```

Comment:
It may not matter for these specific tests, but when testing using a stored payment method, we should be passing the `store` response's `.authorization` rather than a specific param.
So this should work:
```suggestion
    stripe_token = store.params['token']['id']
    auth = @gateway.authorize(@amount, stripe_token, options)
```

Reply:
For Stripe, this is relevant if the PM has an attached `customer`: https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/stripe.rb#L693-L703